airflow/models/taskinstance.py

```diff
@@ -462,11 +465,9 @@ def clear_task_instances(
     for ti in tis:
         TaskInstanceHistory.record_ti(ti, session)
         if ti.state == TaskInstanceState.RUNNING:
-            if ti.job_id:
```

Comment:
i wonder if it matters that we only set to restarting before when job id was set. not sure the implications.  i guess the current analogous thing to do would be to check if it has heartbeated...

Reply:
not even sure what setting ti state to restarting actually does

Reply:
I thought about this, but I can't think at all how a TI could be running without having a job associated with it. I suspect this has not been possible since pre 1.0 days, if even then.